metamorph/plugins/morph_resultsdb.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/python
+import argparse
+import logging
+import logging.config
+import json
+import time
+import os
+import requests
+
+from metamorph.lib.logging_conf import setup_logging
+
+
+class ResultsDBApiException(Exception):
+    pass
+
+
+class ResultsDBApi(object):
+    """
+    Class to communicate and process data with resultsDB
+    """
+    TIMEOUT_LIMIT = 7200  # Wait 2 hours maximally
+    RESULTSDB_LIMITER = 10
+    MINUTE = 60  # 60 seconds
+
+    def __init__(self, job_names, component_nvr, test_tier, resultsdb_api_url):
+        self.resultsdb_api_url = resultsdb_api_url
+        self.job_names = job_names
+        self.job_names_result = {}
+        self.tier_tag = True
+        self.url_options = {'CI_tier': test_tier, 'item': component_nvr}
+
+    def get_test_tier_metadata(self):
+        """
+        Method get_resultsdb_data it's the main function from which data are received
+
+        :returns -- dictionary where keys are job names and their values are list of queried data
+        """
+        if self.job_names:
+            for job_name in self.job_names:
+                self.job_names_result[job_name] = self.get_resultsdb_data(job_name)
+            return self.job_names_result
+        else:
+            queried_data = self.get_resultsdb_data(limit=self.RESULTSDB_LIMITER)
+            self.job_names_result = self.setup_output_data(queried_data)
+            return self.job_names_result
+
+    def query_resultsdb(self, url, url_options=dict, attempt=3):
+        """
+        This method queries resultsDB with given url and url_option variable
+
+        :param url -- resultdb api url
+        :param url_options -- dictionary of wanted options
+        :param attempt -- number of query tries if some problem occurs
+
+        :returns -- Queried data
+        """
+        logging.debug('Running resultsbd API query with this url: {0} and options {1}'.format(url, url_options))
+        response = requests.get(url, params=url_options)
+        if 300 <= response.status_code <= 400:
+            if attempt > 0:
+                time.sleep(self.MINUTE + self.MINUTE)  # Sleeping for two minutes
```

Comment:
I don't think we need to do math in this case. sleep(self.MINUTE) is good enough time to retry if the first try is unsuccessful.